### PR TITLE
Add a setting to control auto release of OpenSearch Monitors managed index creation block

### PR DESCRIPTION
**Pull request by RS146BIJAY**
### Description
OpenSearch controls (applies and remove)```create_index``` block (when all node are breaching high disk watermark) inside ```DiskThresholdMonitor```. Now, if the user tries to apply this block from their end, it may be possible that ```DiskThresholdMonitor``` removes it (```create_index``` gets removed if any of the node is not breaching high disk watermark). This makes this block to be usable only by OpenSearch monitors. User cannot control when these blocks get applied or removed.

We would suggest to add a configurable setting which will control when these blocks will be removed. This will ensure that when set to false, both user and OpenSearch monitor applies the respective blocks, but removal of the block is controlled by user.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/6269

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdSettings.java [hunk 1]
```diff
@@ -97,6 +97,12 @@ public class DiskThresholdSettings {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+    public static final Setting<Boolean> CLUSTER_CREATE_INDEX_BLOCK_AUTO_RELEASE = Setting.boolSetting(
+        "cluster.blocks.create_index.autorelease",
```
Comment (gbbafna) [inline server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdSettings.java:101]: nit : `cluster.blocks.create_index.auto_release `

**Review by gbbafna — COMMENTED** [review]
LTGM Rishav.

File: server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java [hunk 2]
```diff
@@ -224,39 +222,73 @@ public void testIndexCreateBlockNotAppliedWhenAnyNodesBelowHighWatermark() throw
...
+        );
+
         // Validate if index create block is removed on the cluster
         assertBusy(() -> {
             ClusterState state1 = client().admin().cluster().prepareState().setLocal(true).get().getState();
             assertFalse(state1.blocks().hasGlobalBlockWithId(Metadata.CLUSTER_CREATE_INDEX_BLOCK.id()));
         }, 30L, TimeUnit.SECONDS);
     }
 
+    public void testIndexCreateBlockIsRemovedWhenAnyNodesNotExceedHighWatermarkWithAutoReleaseDisabled() throws Exception {
```
Comment (pranikum) [inline server/src/internalClusterTest/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java:258]: You can try to add this scenario.
1. Index create block added by User with autoReleaseDisabled
2. WaterMarkBreaches and Decider also puts Index create block.
3. User release the index block
4. Check that index create block re applies on the cluster by Decider

**Review by pranikum — APPROVED** [review]

File: server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdSettings.java (hunk 1 above)
Comment (RS146BIJAY) [reply server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdSettings.java]: Addressed the comments.

File: CHANGELOG.md [hunk 3]
```diff
@@ -91,6 +91,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add cancellation of in-flight SearchTasks based on resource consumption ([#5606](https://github.com/opensearch-project/OpenSearch/pull/5605))
 - Add support for ppc64le architecture ([#5459](https://github.com/opensearch-project/OpenSearch/pull/5459))
 - [Segment Replication] Add primary weight factor for balanced primary distribution ([#6017](https://github.com/opensearch-project/OpenSearch/pull/6017))
+- Add a setting to control auto release of OpenSearch managed block ([#6269](https://github.com/opensearch-project/OpenSearch/issues/6269))
```
Comment (gbbafna) [inline CHANGELOG.md:94]: should we write `OpenSearch managed index creation block` ?

Comment (RS146BIJAY) [reply CHANGELOG.md]: Addressed the comment. Please check.

**Pull request merged**